prov/efa/test/efa_unit_test_cq.c

```diff
@@ -286,4 +287,122 @@ void test_rdm_cq_read_failed_poll()
 	efa_unit_test_resource_destruct(&resource);
 }
 
+#if HAVE_EFADV_CQ_EX
+static void rdm_cq_read_unknown_peer_ah(int peer_removed) {
+	struct efa_cq *efa_cq;
+	struct rxr_ep *rxr_ep;
+	struct efa_resource resource = {0};
+	struct rxr_pkt_entry *entry;
+	struct efa_ep_addr raw_addr = {0};
+	size_t raw_addr_len = sizeof(raw_addr);
+	struct rdm_peer *peer;
+	struct fi_cq_data_entry cq_entry;
+	struct fi_cq_err_entry cq_err_entry;
+	struct rxr_eager_msgrtm_hdr base_hdr = {0};
+	struct rxr_req_opt_connid_hdr opt_connid_hdr = {0};
+	uint32_t *connid;
+	struct efadv_cq efadv_cq = {0};
+	int ret;
+
+	efa_unit_test_resource_construct(&resource, FI_EP_RDM);
+
+	rxr_ep = container_of(resource.ep, struct rxr_ep, util_ep.ep_fid);
+	efa_cq = container_of(rxr_ep->rdm_cq, struct efa_cq, util_cq.cq_fid);
+
+	/* Create and register a fake peer */
+	ret = fi_getname(&resource.ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+	raw_addr.qpn = 0; // return value of _read_qp_num
+	raw_addr.qkey = 0x1234;
+
+	if (!peer_removed) {
+		fi_addr_t peer_addr;
+		ret = fi_av_insert(resource.av, &raw_addr, 1, &peer_addr, 0, NULL);
+		assert_int_equal(ret, 1);
+	}
+
+	/* Setup packet entry */
+	entry = rxr_pkt_entry_alloc(rxr_ep, rxr_ep->efa_rx_pkt_pool, RXR_PKT_FROM_EFA_RX_POOL);
```

Comment:
this part of constructing an eager rtm packet is useful, consider move it into a separate utility function.

Reply:
Sure will address.